src/zone/scheduler.h

```diff
@@ -27,6 +27,9 @@ namespace zone {
     class SchedulerImpl {
     public:
 
+        using Priority = std::chrono::high_resolution_clock::time_point;
+        using PriorityTask = std::pair<Priority, std::shared_ptr<Task>>;
+
```

Comment:
do not use std::chrono::high_resolution_clock, as it is possible not mono- increasing at some system. You could change it to steady_clock, yet I am not sure it's resolution high enough.

Reply:
Make sense, change it to sequence number with type of unit64_t to indicate the order of tasks.